packages/loopring_v3/contracts/amm/libamm/AmmTransactionReceiver.sol

```diff
@@ -94,19 +101,110 @@ library AmmTransactionReceiver
             signature.offset := add(signature.offset, 0x20)
         }
         if (txType == AmmData.PoolTxType.JOIN) {
+            ctx.approveAmmUpdates(true);
             S.processJoin(
                 ctx,
                 abi.decode(data, (AmmData.PoolJoin)),
                 signature
             );
+            ctx.approveAmmUpdates(false);
         } else if (txType == AmmData.PoolTxType.EXIT) {
+            ctx.approveAmmUpdates(true);
             S.processExit(
                 ctx,
                 abi.decode(data, (AmmData.PoolExit)),
                 signature
             );
+            ctx.approveAmmUpdates(false);
+        } else if (txType == AmmData.PoolTxType.SET_VIRTUAL_BALANCES) {
+            ctx.approveAmmUpdates(true);
+            ctx.vTokenBalancesL2 = ctx.settings.controller.getVirtualBalances(ctx.tokenBalancesL2, ctx.vTokenBalancesL2);
+            ctx.approveAmmUpdates(false);
+        } else if (txType == AmmData.PoolTxType.DEPOSIT) {
```

Comment:
Brecht, deposit from Pool to L2 is done using the 'depositToL2' method, which is a L1 tx. So why do we still need this 'AmmData.PoolTxType.DEPOSIT' type?